tests/integration/ambient/baseline_test.go

```diff
@@ -197,6 +197,12 @@ func TestServices(t *testing.T) {
 			t.Skip("https://github.com/istio/istio/pull/50182")
 		}
 
+		// Ensure we are not leaking metadata exchange headers
+		if !dst.Config().IsUncaptured() {
+			// Probably we shouldn't do it to uncaptured workloads either but ... not an ambient problem.
```

Comment:
Either this comment or the conditional is wrong I think

Reply:
Sidecar -> uncaptured: leaks. Its "intended behavior" for backwards compat reasons, and has nothing to do with ambient.

What we are testing is _every other case_ does not leak